cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt

```diff
@@ -0,0 +1,312 @@
+      field "points", jmes_path(col_item,"points[*].value.doubleValue")
+    end
+  end
+end
+
+
+###############################################################################
+# Scripts
+###############################################################################
+script "js_request_compute_utilization", type: "javascript" do
+  result "request"
+  parameters "param_google_project"
+  code <<-EOF
+  var end_date = new Date().toISOString()
+  var start_date = new Date(new Date().setDate(new Date().getDate() - 30)).toISOString()
+  request = {
+    "auth": "auth_google",
+    "host": "monitoring.googleapis.com",
+    "path": "/v3/projects/"+param_google_project+"/timeSeries/",
+    "query_params": {
+      "filter": 'metric.type="compute.googleapis.com/instance/cpu/utilization"',
+      "interval.endTime": end_date,
+      "interval.startTime": start_date,
+    }
+  }
+EOF
+end
+
+script "js_request_memory_utilization", type: "javascript" do
+  result "request"
+  parameters "param_google_project"
+  code <<-EOF
+  var end_date = new Date().toISOString()
+  var start_date = new Date(new Date().setDate(new Date().getDate() - 30)).toISOString()
+  request = {
+    "auth": "auth_google",
+    "host": "monitoring.googleapis.com",
+    "path": "/v3/projects/"+param_google_project+"/timeSeries/",
+    "query_params": {
+      "filter": 'metric.type="agent.googleapis.com/memory/percent_used" AND metric.label.state=used',
+      "interval.endTime": end_date,
+      "interval.startTime": start_date,
+    }
+  }
+EOF
+end
+
+script "js_request_disk_utilization", type: "javascript" do
+  result "request"
+  parameters "param_google_project"
+  code <<-EOF
+  var end_date = new Date().toISOString()
+  var start_date = new Date(new Date().setDate(new Date().getDate() - 30)).toISOString()
+  request = {
+    "auth": "auth_google",
+    "host": "monitoring.googleapis.com",
+    "path": "/v3/projects/"+param_google_project+"/timeSeries/",
+    "query_params": {
+      "filter": 'metric.type="agent.googleapis.com/disk/percent_used" AND metric.label.state=used',
+      "interval.endTime": end_date,
+      "interval.startTime": start_date,
+    }
+  }
+EOF
+end
+
+script "js_calculated_utilization", type: "javascript" do
+  result "results"
+  parameters "ds_compute_utilization", "ds_memory_utilization", "ds_disk_utilization"
+  code <<-EOS
+     results = []
+     for ( i =0; i < ds_compute_utilization.length; i++ ){
+       var instance_id = ds_compute_utilization[i].instance_id
+       var points = ds_compute_utilization[i].points
+       var memory_record = _.find(ds_memory_utilization, function(record) { return record.instance_id == instance_id; })
+       if (memory_record != null && memory_record !== undefined) {
+        var memory_points = memory_record.points
+        var memory_average = parseFloat(memory_points.reduce(function(acc, val) { return acc + val; })/memory_points.length).toFixed(2);
```

Comment:
you are calculation avg_memory points while collecting the data_source `field "avg_disk_points", jmes_path(col_item, "points[*].value.doubleValue | avg(@)")`. Why are you not just referring to it?

Reply:
I added those, but did not know what the output of the jmes_path calculation would be if points did not exist. I will update today.